src/plots/cartesian/layout_attributes.js

```diff
@@ -465,6 +465,36 @@ module.exports = {
             'Only has an effect if `anchor` is set to *free*.'
         ].join(' ')
     },
+    categorymode: {
+        valType: 'enumerated',
+        values: [
+            'trace', 'category ascending', 'category descending', 'array'
```

Comment:
Given that the axis.type is category, one might think that categories is how you would add data to that axis. 

Great point :+1: 

That's why I'm thinking that `categoryorder` might be the winner here.